### PR TITLE
refactor: remove create category v1 endpoint

### LINKED ISSUE #145: Update the create category functionality to use the upsert catalog object endpoint
**Is your feature request related to a problem? Please describe.**
Currently, the Square API has endpoints for `v1` and `v2` functionality. In [these](https://github.com/hopdrops/hopdrops/blob/master/functions/accounts.py#L18-L21) lines, we invoke the `v1_items` API. The problem with this API is that you need to iterate over *each* location and create a category for said location. Square offers a convenient `v2` endpoint [upsert catalog object](https://developer.squareup.com/reference/square/catalog-api/upsert-catalog-object) that allows you to create a catalog object of type `CATEGORY` that applies to *all* locations. Instead of issuing multiple requests (depending on the amount of locations the merchant has), a single request can be issued to create the category.

**Describe the solution you'd like**
Use the [upsert catalog object](https://github.com/square/square-python-sdk/blob/master/doc/catalog.md#upsert-catalog-object) from the Python SDK to create a new category.

The benefit of using this endpoint is that we can store a single `category_id` in the DynamoDB table and use that as a query for when we receive an [`inventory.count.updated`](https://developer.squareup.com/docs/inventory-api/webhooks) webhook request.

**Additional context**

**Sample requests:**

**Upsert catalog object**

```bash
curl -X POST \
  -H 'Authorization: Bearer EAAAEFYxu_0YjVdrVyMEQ_T5jPD0kqEJqLaF2USPhrapz6O4xL-Pp4ijqcZsuE6X' \
  -d '{
  "idempotency_key": "24B8E2D4-DCCB-416F-93B3-B9D4F0DDE378",
  "object": {
    "category_data": {
      "name": "hopdrops"
    },
    "id": "#hopdrops",
    "type": "CATEGORY"
  }
}' \
https://connect.squareup.com/v2/catalog/object
```

See #108

### PR DESCRIPTION
**Pull request by jasonwalsh**
Removes the V1 [Create Category](https://developer.squareup.com/reference/square/catalog-api/v1-create-category) endpoint from the `accounts` lambda function.

Instead of persisting the locations in this lambda function, persist them in the `redirect` lambda function.

It might also be wise idea to rename this lambda function from `accounts` to `catalog` to closely resemble the Square API.

The V2 [Upsert Catalog Object](https://developer.squareup.com/reference/square/catalog-api/upsert-catalog-object) API persists the category to *all* locations.

Closes #145

### CODE REVIEW

**Comment by jasonwalsh** [comment]
Sort key added as of 3faf665. See #147 for details.

File: functions/accounts.py [hunk 1]
```diff
@@ -12,33 +11,38 @@ def handler(event, context):
...
-            }
+    idempotency_key = uuid4().hex
+
+    response = square.catalog.upsert_catalog_object({
+        'idempotency_key': idempotency_key,
+        'object': {
+            'category_data': {
+                'name': 'hopdrops'
+            },
+            'id': '#hopdrops',
```
Comment (brh55) [inline functions/accounts.py]: Is that id right? `#hopdrops`

Comment (brh55) [reply functions/accounts.py]: Just to double check, but everything looks good to me!

Comment (jasonwalsh) [reply functions/accounts.py]: Yes, from the Square documentation:

> For creates, the object ID must start with #. The provided ID is replaced with a server-generated ID.

See the [documentation](https://developer.squareup.com/reference/square/catalog-api/upsert-catalog-object) for more details.

Comment (brh55) [reply functions/accounts.py]: Kewl beans, rock and load.

**Pull request merged**